### DISCUSSION TITLE
Object not found: storage().from().download()

### DISCUSSION THREAD

**Discussion by Gabriel-Alexandre** [supabase/supabase#21368] [Questions] [answered]
I'm running the code below and I always get an error related to "Object not found". I have already checked that the paths are correct, that my policies are appropriate and that the connection keys with Supabase are correct, I validated that everything is correct. Even so, I continue to receive the error.

Code that causes error:

const { data, error } = await supabase
  .storage
  .from('documets')
  .download('spreadsheet.xlsx')

Error:

StorageApiError: Object not found
    at eval (webpack-internal:///(rsc)/./node_modules/@supabase/storage-js/dist/module/lib/fetch.js:44:24)
    at processTicksAndRejections (node:internal/process/task_queues:95:5) {
  __isStorageError: true,
  status: 400
}

Note: When I run the code below, I don't get any errors, so I think the error is related to .download().

Code that works:

const { data: path } = await supabase
  .storage
  .from('documets')
  .getPublicUrl('spreadsheet.xlsx')

**Comment by GaryAustin1** [comment, marked as answer]
You probably don't have a policy allowing select (download), or don't meet it.
getPublicUrl does not make a server call.  All it does it create a url with string concatenation.   Only that url is usable with a public bucket.  All other operations are still protected by RLS.

**Reply by Gabriel-Alexandre** [reply]
thank you very much, I added the rule and it worked.
